flytekit/annotated/testing.py

```diff
@@ -35,3 +37,28 @@ def _log(*args, **kwargs):
     t.execute = _log
     yield m
     t.execute = _captured_fn
+
+
+def patch(target: Union[PythonTask, Workflow]):
+    if not isinstance(target, PythonTask) and not isinstance(target, Workflow):
+        raise Exception("Can only use mocks on tasks/workflows declared in Python.")
+
+    def wrapper(test_fn):
+        def new_test(*args, **kwargs):
+            logger.warning(f"Invoking mock method for task: '{target.name}'")
+            m = MagicMock()
+            if isinstance(target, PythonTask):
+                saved = target.execute
+                target.execute = m
```

Comment:
I made a dummy method that logs that the mock has been invoked,  do you want to add that here to? Helps in debugging IMO

Reply:
copied the warning above

Reply:
where is the copy?

Reply:
line 52